internal/formats/common/cyclonedxhelpers/properties.go

```diff
@@ -0,0 +1,64 @@
+package cyclonedxhelpers
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/CycloneDX/cyclonedx-go"
+	"github.com/anchore/syft/syft/pkg"
+)
+
+func Properties(p pkg.Package) *[]cyclonedx.Property {
```

Comment:
Is it correct to deconstruct each package and place its separated `Field -> Value` as a list of properties here?

It does move us towards making sure all information is encoded in the format. The definition of `properties` worries me a little though:

```
Provides the ability to document properties in a name-value store. This provides flexibility to include data not officially supported in the standard without having to use additional namespaces or create extensions. Unlike key-value stores, properties support duplicate names, each potentially having different values.
```

If we put data here that IS supported by the specification is the duplicate data too much noise for the final data shape?

Reply:
Fair question. My take: I think it may be easier to lean on the side of larger documents to keep the implementation simpler. If we run into a problem in the future where there are a lot of duplicate fields we could start adding the complexity of removing them.

Reply:
Currently I have only annotated fields that do not directly map to a cyclonedx field with the `cyclonedx` struct tag to allow us to avoid duplication. The properties logic only includes fields with the `cyclonedx` struct tag.